src/core/Proc/Async.pm6

```diff
@@ -260,11 +272,11 @@ my class Proc::Async {
 
     method !capture(\callbacks,\std,\the-supplier) {
         my $promise = Promise.new;
-        my $vow = $promise.vow;
+        the-supplier.vow = $promise.vow;
```

Comment:
This seems race-y ( although I haven't bothered to investigate that hunch )

Reply:
Looking at it again, you're right. I figure this particular method is probably alright (since it's a private method called with 3 different suppliers each time), the 'start' method could definitely cause a data race if called multiple times from different threads. I've updated the branch to add a commit wrapping it in a Lock.

I've cherry-picked that commit and will submit it on it's own, so that it can be considered regardless of what we decide to do with this PR.